MOTIVATION.md

```diff
@@ -0,0 +1,138 @@
+
+Analysis package developers should be able to integrate their existing work with
+HCA data:
+
+```R
+library(HumanCellAtlas)
+library(SingleCellExpression)
+
+filtered.data <- hca %>%
+    filter(percent_mito < 0.05) %>%
+    filter(organ == "kidney")
+
+sce <- as(filtered.data, "SingleCellExperiment")
+...
+```
+
+And portal and other interactive tool developers should be able to have their
+applications make requests for HCA data with minimal overhead using web best
+practices.
+
+# Challenges
+
+A number of engineering challenges need to be addressed before the interactions
+sketched above can be realized.
+
+## Scientific Utility
+
+#### Normalization and batch effect correction.
+
+The DCP combines single-cell data from many different sources. But, batch
+effects in single-cell data are particularly strong. Correct analysis of
+combined data from the HCA requires careful handling of batch and other
+normalization issues. The matrix service could prepare a matrix from multiple
+experiments that is simple and easy to use, but such a matrix may only serve to
+enable false discoveries.
+
+#### Metadata availability.
+
+Much of the engineering work around expression matrices focuses on storing and
+delivering the expression values themselves, but practical analysis of
+expression data often turns much more on metadata. An analyst needs to be able
+to access and operate on annotations of cells, genes, and experiments.
+
+#### Data formats and schemata
+
+Multiple single-cell analysis tools have been developed that depend on
+different representations of expression matrix data and metadata. The
+differences limit interoperability and the matrix service's ability to
+integrate seamlessly many tools in current or future use. A limited set of
+dataframe standards agreed to by the community will be needed.
+
+## Scale
+
+The number of cells with expression data in the HCA DCP will grow by several
+orders of magnitude over time. The matrix service will need to handle this
+growth in data volume.
+
+#### Query responsiveness
+
+The matrix service delivers subsets of HCA expression data in response to
+queries from users. As the HCA and the complexity of supported queries grows,
+the time needed to deliver a matrix could become intolerably long. The matrix
+service will need a scalable, burstable architecture.
+
+#### Distributed computation
+
+Most current tertiary analyses of single-cell expression data can be reasonably
+performed on a single computer, but this will change soon. The matrix service
+should enable analyses distributed computation over expression data.
+
+#### Data representation and localization
+
+As data volume grows, when and how to localize expression data to a client will
+become important considerations. For example, a client should be able to refer
+to subsets or operations on HCA data without unexpectedly localizing a huge
+amount expression data.
+
+## Expressing Queries
+
+#### Separation of query and matrix interfaces
+
```

Comment:
This surprises me, since my understanding is that the matrix service is providing a set of requirements to the query service? Wouldn't we fail if the matrix service was misaligned?

Reply:
I'm thinking more about user interface rather than capabilities here. The query service is quite likely to expose SQL or something SQL-like for queries. Ultimately, we'd prefer DCP data consumers to use something more natural for them, like pandas-like queries or whatever people do in R.

Reply:
That clarifies a bit - more of a translation or rewrite of queries perhaps similar to [How to rewrite your SQL queries in Pandas, and more](https://codeburst.io/how-to-rewrite-your-sql-queries-in-pandas-and-more-149d341fc53e) -